git/git.go

```diff
@@ -81,15 +81,27 @@ func CurrentBranch() (string, error) {
 }
 
 func CurrentRemoteRef() (*Ref, error) {
-	remote, err := CurrentRemote()
+	remoteref, err := RemoteRefNameForCurrentBranch()
 	if err != nil {
 		return nil, err
 	}
 
-	return ResolveRef(remote)
+	return ResolveRef(remoteref)
 }
 
-func CurrentRemote() (string, error) {
+// RemoteForCurrentBranch returns the name of the remote that the current branch is tracking
+func RemoteForCurrentBranch() (string, error) {
+	ref, err := RemoteRefNameForCurrentBranch()
+	if err != nil {
+		return "", err
+	}
+	// ref name is full name, split first part for remote
+	parts := strings.Split(ref, "/")
+	return parts[0], nil
```

Comment:
Wait, what? :scream: I'm sure that used to be disallowed. OK, I'll have to cope with that then...